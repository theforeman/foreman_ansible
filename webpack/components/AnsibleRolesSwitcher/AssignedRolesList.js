import React from 'react';
import { ListView } from 'patternfly-react';
import PaginationWrapper from 'foremanReact/components/Pagination/PaginationWrapper';
import { reject } from 'lodash';

import AnsibleRole from './AnsibleRole';

class AssignedRolesList extends React.Component {
  render() {
    const {
      assignedRoles,
      pagination,
      itemCount,
      onPaginationChange,
      onRemoveRole,
      resourceName,
    } = this.props;

    const directlyAssignedRoles = reject(assignedRoles, role => role.inherited);

    return (
      <div>
        <ListView>
          <div className="sticky-pagination sticky-pagination-grey">
            <PaginationWrapper
              viewType="list"
              itemCount={itemCount}
              pagination={pagination}
              onChange={onPaginationChange}
              dropdownButtonId='assigned-ansible-roles-pagination-row-dropdown'
            />
          </div>
          { assignedRoles.map(role => <AnsibleRole key={role.id}
                                                   role={role}
                                                   icon='fa fa-minus-circle'
                                                   onClick={onRemoveRole}/>) }
        </ListView>
        <div>
          { directlyAssignedRoles.map(role => <input key={role.id}
                                                     type="hidden"
                                                     name={`${resourceName}[ansible_role_ids][]`}
                                                     value={role.id} />) }
        </div>
      </div>
    );
  }
}

export default AssignedRolesList;
