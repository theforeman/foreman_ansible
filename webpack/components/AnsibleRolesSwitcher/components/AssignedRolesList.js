import React from 'react';
import { ListView } from 'patternfly-react';
import PaginationWrapper from 'foremanReact/components/Pagination/PaginationWrapper';
import { isEmpty } from 'lodash';
import PropTypes from 'prop-types';

import AnsibleRole from './AnsibleRole';

const AssignedRolesList = ({
  assignedRoles,
  pagination,
  itemCount,
  onPaginationChange,
  onRemoveRole,
  resourceName,
}) => {
  const directlyAssignedRoles = assignedRoles.filter(role => !role.inherited);

  return (
    <div>
      <ListView>
        <div className="sticky-pagination sticky-pagination-grey">
          <PaginationWrapper
            viewType="list"
            itemCount={itemCount}
            pagination={pagination}
            onChange={onPaginationChange}
            dropdownButtonId="assigned-ansible-roles-pagination-row-dropdown"
          />
        </div>
        {assignedRoles.map(role => (
          <AnsibleRole
            key={role.id}
            role={role}
            icon="fa fa-minus-circle"
            onClick={onRemoveRole}
            resourceName={resourceName}
          />
        ))}
      </ListView>
      <div>
        {isEmpty(directlyAssignedRoles) ? (
          <input
            type="hidden"
            name={`${resourceName}[ansible_role_ids][]`}
            value=""
          />
        ) : (
          directlyAssignedRoles.map(role => (
            <input
              key={role.id}
              type="hidden"
              name={`${resourceName}[ansible_role_ids][]`}
              value={role.id}
            />
          ))
        )}
      </div>
    </div>
  );
};

AssignedRolesList.propTypes = {
  assignedRoles: PropTypes.arrayOf(PropTypes.object).isRequired,
  pagination: PropTypes.shape({
    page: PropTypes.number,
    perPage: PropTypes.number,
  }).isRequired,
  itemCount: PropTypes.number.isRequired,
  onPaginationChange: PropTypes.func.isRequired,
  onRemoveRole: PropTypes.func.isRequired,
  resourceName: PropTypes.string.isRequired,
};

export default AssignedRolesList;
