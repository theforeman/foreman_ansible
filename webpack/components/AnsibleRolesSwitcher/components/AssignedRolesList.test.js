import { testComponentSnapshotsWithFixtures } from 'react-redux-test-utils';

import AssignedRolesList from './AssignedRolesList';

const noop = () => {};

const fixtures = {
  'should render': {
    assignedRoles: [{ id: 1, name: 'fake.role' }, { id: 2, name: 'test.role' }],
    pagination: { page: 1, perPage: 25 },
    itemCount: 2,
    onPaginationChange: noop,
    onRemoveRole: noop,
    resourceName: 'host',
  },
};

describe('AssignedRolesList', () => testComponentSnapshotsWithFixtures(AssignedRolesList, fixtures));
