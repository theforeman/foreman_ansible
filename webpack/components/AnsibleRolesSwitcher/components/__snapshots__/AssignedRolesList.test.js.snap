// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AssignedRolesList should render 1`] = `
<div>
  <ListView
    className=""
  >
    <div
      className="sticky-pagination sticky-pagination-grey"
    >
      <PaginationWrapper
        dropdownButtonId="assigned-ansible-roles-pagination-row-dropdown"
        itemCount={2}
        onChange={[Function]}
        pagination={
          Object {
            "page": 1,
            "perPage": 25,
          }
        }
        viewType="list"
      />
    </div>
    <AnsibleRole
      icon="fa fa-minus-circle"
      key="1"
      onClick={[Function]}
      resourceName="host"
      role={
        Object {
          "id": 1,
          "name": "fake.role",
        }
      }
    />
    <AnsibleRole
      icon="fa fa-minus-circle"
      key="2"
      onClick={[Function]}
      resourceName="host"
      role={
        Object {
          "id": 2,
          "name": "test.role",
        }
      }
    />
  </ListView>
  <div>
    <input
      key="1"
      name="host[ansible_role_ids][]"
      type="hidden"
      value={1}
    />
    <input
      key="2"
      name="host[ansible_role_ids][]"
      type="hidden"
      value={2}
    />
  </div>
</div>
`;
